File: 1. Easy/ASCII ART/Solution.cpp
#include <string>
#include <cctype>
#include <iostream>

using namespace std;

int main()
{
    int WIDTH, HEIGHT, shift;
    string TEXT, line;

    cin >> WIDTH >> HEIGHT;
    getline(cin, TEXT);
   
    for (int i = 0; i < HEIGHT; ++i) {
        getline(cin, line);
        
        for (const char& symbol: TEXT) {
            shift = isalpha(symbol) ? toupper(symbol) - 65 : 26;
            cout << line.substr(WIDTH * shift, WIDTH);
        }
        
        cout << endl;
    }
}

File: 1. Easy/ASCII ART/Solution.py
import sys
import math

def printChar(line, char, row, width):
    begin = (ord(char) - ord('a')) * width
    end = begin + width
    print(row[line][begin:end], end = "")

width = int(input())
height = int(input())
text = input().lower()
row = []
for i in range(height):
    row.append(input())
    
for i in range(height):
    for j in text:
        if (ord('a') <= ord(j) <= ord('z')):
            printChar(i, j, row, width)
        else:
            printChar(i, chr(ord('z') + 1), row, width)
    print()